floris/tools/wind_data.py

```diff
@@ -0,0 +1,529 @@
+# Copyright 2024 NREL
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# See https://floris.readthedocs.io for documentation
+
+import matplotlib.cm as cm
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from pandas.api.types import CategoricalDtype
+
+
+# Define the super lass that WindRose and TimeSeries inherit
+# Define functions here that are either the same for both WindRose and
+# TimeSeries or will be overloaded
+class WindDataBase:
+    def __init__():
+        pass
+
+    def unpack_for_reinitialize(self):
+        """
+        Return only the variables need for reinitialize
+        """
+        (
+            wind_directions_unpack,
+            wind_speeds_unpack,
+            _,
+            ti_table_unpack,
+            _,
+        ) = self.unpack()
```

Comment:
This is actually doing the opposite of the usual inheritance. The super-class is calling a function that's on the sub-class. At a minimum, there should be an `unpack` function on `WindDataBase` that raises a `NotImplementedError` if it's called.

Reply:
Thanks for the point here. I've now add an `unpack()` method to the base class that raises a `NotImplementedError` as you suggest, and also added a test to check that works as expected.